src/core/Pipelining.ts

```diff
@@ -90,11 +83,16 @@ export class Pipelining<E> {
      *
      * If not, it will return void when all of the requests are completed.
      *
-     * @throws {Error} if any of the requests fails
+     * @throws {Error} with the error of the failing request if any of the requests fail. Also,
+     *  subsequent calls to this method with the same pipelining instance rejects.
      * @returns array of the results or void
      */
     run(): Promise<void | E[]> {
-        this.reset();
+        if (this.isPipeliningRan) {
+            return Promise.reject('This pipelining instance was already run. ' +
```

Comment:
It seems to me like a good idea to mention this behavior (rejection on non-first `run()`) in JSDoc and readme. WDYT?

Reply:
Thanks for the review. I will clarify this behaviour on documentations